### PR TITLE
Feature/tests

### PR DESCRIPTION
**Pull request by hamidazim321**
## Progress Summary
- Me and my partner have added snapshot tests for each component
- We have added tests by simulating user interaction for the App component
- We have added tests for the Quotes component to ensure it displays error when fetch request fails and quote when it passes

## Changes Made after the recent previous review
- I have added tests for the dot button on the calculator in operate, calculate function and in Calculator component
- I have updated the link to the issues page on the Readme file

### CODE REVIEW

File: src/__tests__/Calculate.test.js
Comment (cynthiainga) [inline src/__tests__/Calculate.test.js:1]: - [ ] Good job so far but as mentioned in the project requirements you should test all the functionalities & scenarios:
  - The `'0'` button should be tested,
  - The `'.'` button should be tested,
  - The `'+/-'` button should be tested as well,

Kindly fix that 🙃 :wink'

File: src/__tests__/Operate.test.js
Comment (cynthiainga) [inline src/__tests__/Operate.test.js:1]: - [ ] The `'%'` operation should be tested 🙃 
- [ ] Also you should test the scenario when the `÷` and `%` operations return the error `Math Error` 😁 😉

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @hamidazim321 & @Wineshuga, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] All implemented tests are passing ✔️ 
- [ ] Good commit history ✔️ 
- [ ] No linters errors ✔️ 

## Required Changes ♻️

- [ ] The link to the issue page in the README file is broken kindly fix that, having broken links in the README file doesn’t look more professional 😎 😎 
- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:
- I think it would be nice to add @Wineshuga's contact info to the README file as she contributed to this project 😁 😁 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "react-dom": "^18.2.0",
     "react-router-dom": "^6.15.0",
     "react-scripts": "5.0.1",
+    "react-test-renderer": "^18.2.0",
     "web-vitals": "^2.1.4"
   },
```
Comment (addod19) [inline package.json:17]: - [ ] As mentioned in the previous review, the link to the issues page is broken and needs to be fixed

File: src/__tests__/Calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import calculate from '../logic/Calculate';
+
```
Comment (addod19) [inline src/__tests__/Calculate.test.js:2]: - [ ] As mentioned by the previous reviewer, let us kindly test for the button `'.'`

**Review by addod19 — CHANGES_REQUESTED** [review]
## Hi @hamidazim321 and @Wineshuga, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Nice job addressing some of the comments from the previous review :rocket: 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (hamidazim321) [reply package.json:17]: @addod19 Sorry I don't understand what you are asking me to change. You are highlighting the react-test-renderer dependency in the package.json file but mentioning link to the issues page. Are you talking about the link to the issues page on my README.md file?

**Review by KDlamini — APPROVED** [review]
### Hi team @hamidazim321, @Wineshuga 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the requirements for writing unit tests for the Math Magicians app. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Good PR title and description
✔️ The [Jest](https://jestjs.io/) and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) libraries are used
✔️ Functionalities & scenarios of the `operate.js` and `calculate.js` modules are tested
✔️ Jest snapshots for testing all the components and user interaction simulation implemented
✔️ Well-documented and professional README

# Status: Approved ✔️ 

LeonardoInGatsby
 
Cheers 🥂  and Happy coding!!! 👯 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**